sources/packages/api/src/route-controllers/student/student.students.controller.ts

```diff
@@ -60,19 +81,110 @@ export class StudentStudentsController extends BaseController {
     private readonly gcNotifyActionsService: GCNotifyActionsService,
     private readonly applicationService: ApplicationService,
     private readonly studentService: StudentService,
+    private readonly formService: FormService,
+    private readonly atbcService: ATBCService,
+    private readonly studentRestrictionService: StudentRestrictionService,
   ) {
     super();
   }
 
+  /**
+   * Creates the student checking for an existing user to be used or
+   * creating a new one in case the user id is not provided.
+   * The user could be already available in the case of the same user
+   * was authenticated previously on another portal (e.g. parent/partner).
+   * @param payload information needed to create the user.
+   */
+  @Post()
+  @ApiUnprocessableEntityResponse({
+    description:
+      "There is already a student associated with the user or the request is invalid.",
+  })
+  @RequiresStudentAccount(false)
+  async create(
+    @UserToken() studentUserToken: StudentUserToken,
+    @Body() payload: CreateStudentAPIInDTO,
+  ): Promise<void> {
```

Comment:
For Post APIs we are returning the primary identifier of the created resource.

Reply:
I missed that while moving the code. I will add it.